File: 프로그래머스/3/43162. 네트워크/네트워크.java
class Solution {
    public int solution(int n, int[][] computers) {
        boolean[] visited=new boolean[n];
        int answer = 0;
        for(int i=0;i<n;i++){
            if(!visited[i]){
                dfs(n,computers,visited,i);
                answer++;
            }
        }
        return answer;
    }
    
    private void dfs(int N, int[][] computers, boolean[] visited,int u){
        visited[u]=true;
        for (int v=0;v<N;v++){
            if (computers[u][v]==1&&!visited[v]){
                dfs(N,computers,visited,v);
            }
        }
    }
}

File: 프로그래머스/3/43162. 네트워크/네트워크.py
def solution(N, edges):
    
    # INIT
    visited=[False]*N
    
    def dfs(u):
        visited[u]=True
        for v in range(N):
            if edges[u][v] and not visited[v]:
                dfs(v)
    
    count=0
    for i in range(N):
        if not visited[i]:
            count+=1
            dfs(i)
    
    return count